examples/arrays/arrays.md

```diff
@@ -1,7 +1,17 @@
 # Arrays
 
-`T[]` is an `array` of `T`. Arrays are mutable. `==` and `!=` on arrays is deep: two arrays are equal if they have the same members in the same order. Ordering is lexicographical based on the ordering of the members. Langlib `arr.length()` function gets the length; `arr.setLength(n)` sets the length.
+An array can be used to hold set of values of a same type `T`. Array type can be defined as `T[n]` where, `T` is the element type and `n` is the length of the array. `n` must be an integer literal or constant reference of type int. Optionally we can create a variable length array by defining an array without `n` as `T[]`.
+
+The length of the array can be inferred from the context by defining array as `T[*]`. Length of the array should be known in compile time.
```

Comment:
Do we need to mention that the inferring is supported only in 1st dimension?

Reply:
Will add in the nested array BBE